torchvision/datasets/lsun.py

```diff
@@ -75,27 +77,28 @@ def __init__(self, root, classes='train', transform=None, target_transform=None)
                       'living_room', 'restaurant', 'tower']
         dset_opts = ['train', 'val', 'test']
 
-        if type(classes) == str and classes in dset_opts:
+        try:
+            verify_str_arg(classes, dset_opts, "classes")
             if classes == 'test':
                 classes = [classes]
             else:
                 classes = [c + '_' + classes for c in categories]
-        elif type(classes) == list:
-            for c in classes:
-                c_short = c.split('_')
-                c_short.pop(len(c_short) - 1)
-                c_short = '_'.join(c_short)
-                if c_short not in categories:
-                    raise (ValueError('Unknown LSUN class: ' + c_short + '.'
-                                      'Options are: ' + str(categories)))
-                c_short = c.split('_')
-                c_short = c_short.pop(len(c_short) - 1)
-                if c_short not in dset_opts:
-                    raise (ValueError('Unknown postfix: ' + c_short + '.'
-                                      'Options are: ' + str(dset_opts)))
-        else:
-            raise (ValueError('Unknown option for classes'))
-        self.classes = classes
+        except ValueError:
+            # TODO: Should this check for Iterable instead of list?
+            if isinstance(classes, list):
+                for c in classes:
+                    # TODO: This assumes each item is a str (or subclass). Should this
+                    #   also be checked?
+                    c_short = c.split('_')
+                    category, dset_opt = '_'.join(c_short[:-1]), c_short[-1]
+                    # TODO: "LSUN class" and "postfix" are not arguments.
+                    #  verify_str_arg needs to be adapted to reflect this
```

Comment:
I'm not sure I follow, can you explain a bit more?

Reply:
The error would read: "Unknown value {} for argument LSUN class.". I think this might be confusing, since LSUN class is not an argument. 

I would solve this with your proposal from below: overriding the default message if custom is given.

Reply:
sounds good